extension/content/components/recipes/form/filters/FallbackFO.js

```diff
@@ -0,0 +1,83 @@
+// @ts-nocheck
+import React from "react";
+import { Badge, ControlLabel, FormGroup } from "rsuite";
+
+import JsonEditor from "devtools/components/common/JsonEditor";
+import {
+  ACTION_UPDATE_DATA,
+  useRecipeDetailsData,
+  useRecipeDetailsDispatch,
+} from "devtools/contexts/recipeDetails";
+
+const KNOWN_FILTER_TYPES = [
+  "channel",
+  "version",
+  "country",
+  "locale",
+  "bucketSample",
+  "stableSample",
+];
+
+export default function FallbackFO() {
+  const data = useRecipeDetailsData();
+  const dispatch = useRecipeDetailsDispatch();
+  const [invalidJSON, setInvalidJson] = React.useState(false);
+
+  let knownFO = [];
+  let additionalFO = [];
+  if (data.filter_object) {
+    [knownFO, additionalFO] = data.filter_object.reduce(
+      ([knownFO, additionalFO], fo) => {
+        if (fo && KNOWN_FILTER_TYPES.includes(fo.type)) {
+          return [[...knownFO, fo], additionalFO];
+        }
+
+        return [knownFO, [...additionalFO, fo]];
+      },
+      [[], []],
+    );
+  }
+
+  const handleChange = (value, err) => {
+    if (value && !err) {
+      const newFilterObjects = [...knownFO, ...value];
+
+      dispatch({
+        type: ACTION_UPDATE_DATA,
+        data: {
+          ...data,
+          filter_object: newFilterObjects,
+        },
+      });
+      setInvalidJson(false);
+    } else if (value === "") {
+      const newFilterObjects = [...knownFO];
+
+      dispatch({
+        type: ACTION_UPDATE_DATA,
+        data: {
+          ...data,
+          filter_object: newFilterObjects,
+        },
+      });
+      setInvalidJson(false);
+    } else {
+      setInvalidJson(true);
+    }
+  };
```

Comment:
i think there's room to refactor this to be more DRY. also there is the issue of what if `value` is not an array. i think this fits into the broader issue of surfacing form errors better and we should probably have a discussion about this.

Just to address the DRY bits:
```js
const handleChange = (value, err) => {
  if (err) {
    setInvalidJson(true);
  } else {
    setInvalidJson(false);
    let newFilterObjects = [...knownFo];
    if (value) {
      newFilterObjects = newFilterObjects.concat(value);
    }
    dispatch({
      type: ACTION_UPDATE_DATA,
      data: {
        ...data,
        filter_object: newFilterObjects,
      },
    });
  }
}
```

Reply:
Made the suggested changes. When you say surfacing form errors, not sure if you meant the fallback editor specifically or all the other forms as well or whether changes should be done within this PR?

Reply:
I don't think we should handle all the form errors in this PR but I think it's prudent to come up with a plan for dealing with all form errors (I think there are two classes of errors: server-side and client-side validation errors) that we need to deal with and we should address this client-side error with a broader framework in mind.